### PR TITLE
fix: move build into mgmake package

### PR DESCRIPTION
**Pull request by viktorvoltaire**

### CODE REVIEW

File: .mage/magefile.go [hunk 1]
```diff
@@ -31,8 +31,8 @@ import (
 )
 
 func init() {
-	mgmake.GenerateMakefiles(
-		mgmake.Makefile{
+	gen.GenerateMakefiles(
```
Comment (odsod) [inline .mage/magefile.go:34]: IMO we should find a better package name for this public API

Comment (viktorvoltaire) [reply .mage/magefile.go]: I reverted to the old package name

File: example/.mage/magefile.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ package main
 
 import (
 	"github.com/magefile/mage/mg"
-	"go.einride.tech/mage-tools/mgmake"
+	"go.einride.tech/mage-tools/mgbuild/mgmake"
```
Comment (odsod) [inline example/.mage/magefile.go:8]: does `mgmake` need to be a child, or can we maintain the flat structure by putting `mgmake` at the same level?

File: mgbuild/main.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package main
```
Comment (odsod) [inline mgbuild/main.go:1]: Non top-level commands are usually put under the `cmd` subdirectory

Comment (odsod) [reply mgbuild/main.go:1]: `go run go.einride.tech/mage-tools/cmd/build`

File: example/.mage/magefile.go (hunk 2 above)
Comment (viktorvoltaire) [reply example/.mage/magefile.go:8]: It does not

File: mgbuild/main.go (hunk 3 above)
Comment (viktorvoltaire) [reply mgbuild/main.go:1]: :+1:

File: example/.mage/magefile.go (hunk 2 above)
Comment (odsod) [reply example/.mage/magefile.go:8]: Disregard me - since it's a command let's go with the comment below!

File: mgbuild/main.go (hunk 3 above)
Comment (viktorvoltaire) [reply mgbuild/main.go]: Fixed

**Pull request merged**